caldav/server.go

```diff
@@ -365,7 +367,18 @@ func (b *backend) Propfind(r *http.Request, propfind *internal.Propfind, depth i
 		resps = append(resps, *resp)
 
 		if depth != internal.DepthZero {
-			// TODO
+			cos, err := b.Backend.ListCalendarObjects(r.Context(), &compReq)
```

Comment:
Hm, right. It _seems_ like an empty `<address-data>` element is semantically equal to `<allprop>`. In that case it sounds simpler to set `AllProp = true` to simplify logic in backends?

Reply:
do you mean just here, or in the unmarshalling of `address-data` in general? The backends will still have to handle all cases unless we do the latter, right?

Reply:
Yeah, I mean in `decodeCalendarDataReq` and `decodeAddressDataReq`.

Reply:
It seems we've reached the depths of these RFCs where the fun truly starts :wink: 

First, we have to make sure to distinguish between `DAV:allprop` and e.g. `caldav:allprop`. Setting these actually limits the properties to be returned to those defined in the the respective RFC. Sine we are talking about a plain WebDAV PROPFIND here, it would be `DAV:allprop`. If this is set, pretty much all of the CalDAV properties SHOULD NOT be returned according to the  RFC. So setting allprop is definitely semantically different from an empty query (for both WebDAV and CalDAV).

We also cannot translate the `internal.Propfind` into a `caldav.CalendarCompRequest`, because that would (wrongly) translate the `allprop` from a `DAV:allprop` to a `caldav:allprop`.

Currently, the filtering of returned properties happens after all properties are collected, in `internal.NewPropfindResponse`. As I see it,  it would currently be best to leave this as is, putting potentially a bit of unneeded load on the backend. Going forward, we _could_ add a code path for when `DAV:allprop` is set to only get the properties from the backend that we need to respond with basic DAV properties. But I am not convinced it is worth it. And I am fairly sure attempting to translate all possible values for `prop` and `include` in the `Propfind` to optimized queries to the backend would totally blow up complexity.